File: backend/app/schemas/guitai_song.py
from typing import Literal

from pydantic import BaseModel


class SpotifyTrackInput(BaseModel):
    id: str
    title: str
    artist: str
    album: str
    image_url: str | None = None
    spotify_url: str | None = None
    duration_ms: int
    explicit: bool


class GuitAISongResponse(BaseModel):
    id: str

    source: Literal["spotify"]

    source_id: str

    title: str
    artist: str
    album: str

    image_url: str | None = None

    source_url: str | None = None

    duration_ms: int

    explicit: bool

    status: Literal[
        "waiting_for_lesson"
    ]

    lesson_available: bool

File: src/types/guitaiSong.ts
export type GuitAISongStatus = "waiting_for_lesson";

export type GuitAISong = {
  id: string;

  source: "spotify";

  source_id: string;

  title: string;
  artist: string;
  album: string;

  image_url: string | null;

  source_url: string | null;

  duration_ms: number;

  explicit: boolean;

  status: GuitAISongStatus;

  lesson_available: boolean;
};
